partiql-logical/src/lib.rs

```diff
@@ -108,6 +108,55 @@ pub enum PathComponent {
     Index(i64),
 }
 
+#[derive(Clone, Debug)]
+pub struct IsTypeExpr {
+    pub not: bool,
+    pub expr: Box<ValueExpr>,
+    pub is_type: Type,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Type {
```

Comment:
`Type` copied from `ast.rs`. Currently used in both the logical plan and physical plan. Perhaps a better place would be the `partiql-types` crate?